core/sparse_array.cpp

```diff
@@ -95,6 +141,31 @@ void SparseArray::Reset(unsigned int p_nrows, unsigned int p_ncols)
 	finished_ = false;
 }
 
+void SparseArray::Resize(uint32_t p_row)
+{
+    //vectors are reallocated by 1000 entries at a time. It can be wasteful but probably better than making a function call for each new entry. Will defer the final decision to Ben. 
+
+
+    *(columns + p_row)=(uint32_t *)realloc(*(columns + p_row), 1000 * sizeof(uint32_t));  
+    *(distances + p_row)=(uint32_t *)realloc(*(distances + p_row), 1000 * sizeof(uint32_t));
+
+    *(nnz_capacity + p_row)+= 1000; //Update capacity 
+}
+
+void SparseArray::AddDistance(uint32_t p_row, const uint32_t p_column, sa_distance_t p_distance)
```

Comment:
I'm not sure whether the compiler is already set up to inline methods that aren't marked explicitly as inline, but this is obviously a prime candidate for inlining; might move it to the header and make it explicitly inline to encourage dumb compilers to do the right thing.